### PR TITLE
[DMS-286] Docker Compose improvements for a complete environment

### PR DESCRIPTION
**Pull request by stephenfuqua**

### CODE REVIEW

File: src/backend/EdFi.DataManagementService.Backend.Postgresql/Deploy/Scripts/0004.3_Configure_Replication.sql [hunk 1]
```diff
@@ -0,0 +1,6 @@
+-- SPDX-License-Identifier: Apache-2.0
+-- Licensed to the Ed-Fi Alliance under one or more agreements.
+-- The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+-- See the LICENSE and NOTICES files in the project root for more information.
+
+ALTER PUBLICATION to_debezium ADD TABLE dms.document;
```
Comment (stephenfuqua) [inline src/backend/EdFi.DataManagementService.Backend.Postgresql/Deploy/Scripts/0004.3_Configure_Replication.sql:6]: Separated into three files in order to have separate transactions.

File: src/backend/EdFi.DataManagementService.Backend.Postgresql/Operation/GetDocumentById.cs [hunk 2]
```diff
@@ -52,7 +52,6 @@ NpgsqlTransaction transaction
                 return new GetResult.GetFailureNotExists();
             }
 
-            // TODO: Documents table needs a last modified datetime
```
Comment (stephenfuqua) [inline src/backend/EdFi.DataManagementService.Backend.Postgresql/Operation/GetDocumentById.cs:55]: We have a ticket to deal with this, no need for a TODO comment (even if I'm the one who wrote it). The code analyzer was _sometimes_ (strangely, not _always_) complaining about these TODO comments.

File: src/core/EdFi.DataManagementService.Core/Middleware/ValidateQueryMiddleware.cs [hunk 3]
```diff
@@ -78,7 +78,7 @@ out bool totalValue
             }
         }
 
-        // TODO: DMS-312 Validate all other query parameters are valid ones for the current resource
+        // DMS-312 Validate all other query parameters are valid ones for the current resource
```
Comment (stephenfuqua) [inline src/core/EdFi.DataManagementService.Core/Middleware/ValidateQueryMiddleware.cs:81]: Whoever works on DMS-312 might be looking for this comment, therefore I left it in place though without "TODO".

File: src/run.sh [hunk 4]
```diff
@@ -9,7 +9,11 @@ set +x
 
 envsubst < /app/appsettings.template.json > /app/appsettings.json
 
-until pg_isready -h ${POSTGRES_HOST} -p ${POSTGRES_PORT} -U ${POSTGRES_ADMIN_USER}; do
+# The following command effectively splits the connection string into separate variables
+eval ${DATABASE_CONNECTION_STRING_ADMIN}
```
Comment (stephenfuqua) [inline src/run.sh:13]: This is probably a dangerous thing to do, but it _is_ very effective.

File: src/backend/EdFi.DataManagementService.Backend.OpenSearch/QueryOpenSearch.cs [hunk 5]
```diff
@@ -108,6 +108,11 @@ ILogger logger
 
             JsonNode hits = JsonSerializer.Deserialize<JsonNode>(response.Body)!["hits"]!;
 
+            if (hits is null)
+            {
+                return new QueryResult.QuerySuccess(new JsonArray(), 0);
+            }
+
```
[the remark this answers is not in the document]
Comment (stephenfuqua) [reply src/backend/EdFi.DataManagementService.Backend.OpenSearch/QueryOpenSearch.cs:115]: This change cause unit test coverage to fail 👎

File: src/core/EdFi.DataManagementService.Core.Tests.Unit/ApiSchema/Extensions/JsonHelperExtensionTests.cs
Comment (stephenfuqua) [inline src/core/EdFi.DataManagementService.Core.Tests.Unit/ApiSchema/Extensions/JsonHelperExtensionTests.cs:1]: A bunch of unit tests to drive up code coverage % enough to pass inspection.

Comment (bradbanister) [reply src/core/EdFi.DataManagementService.Core.Tests.Unit/ApiSchema/Extensions/JsonHelperExtensionTests.cs:1]: 🤯

Comment (stephenfuqua) [reply src/core/EdFi.DataManagementService.Core.Tests.Unit/ApiSchema/Extensions/JsonHelperExtensionTests.cs:1]: .. and it still wasn't enough. I added 1 `if` statement, and a bunch of tests that cover branches, and it didn't solve it. This strikes me as very strange.

File: src/tests/EdFi.DataManagementService.Tests.E2E/Management/ContainerSetup.cs [hunk 6]
```diff
@@ -45,23 +47,19 @@ public static async Task<string> SetupDataManagement()
...
-            )
-            .WithEnvironment("POSTGRES_ADMIN_USER", pgAdminUser)
-            .WithEnvironment("POSTGRES_ADMIN_PASSWORD", pgAdminPassword)
-            .WithEnvironment("POSTGRES_PORT", "5432")
-            .WithEnvironment("POSTGRES_HOST", dbContainerName)
+            .WithEnvironment("DATABASE_CONNECTION_STRING", connectionString)
+            .WithEnvironment("DATABASE_CONNECTION_STRING_ADMIN", connectionString)
             .WithEnvironment("LOG_LEVEL", "Debug")
-            .WithEnvironment("OAUTH_TOKEN_ENDPOINT", "http://127.0.0.1:8080/oauth/token")
+            .WithEnvironment("OAUTH_TOKEN_ENDPOINT", $"http://127.0.0.1:8080/oauth/token")
```
Comment (stephenfuqua) [inline src/tests/EdFi.DataManagementService.Tests.E2E/Management/ContainerSetup.cs:53]: Oops, that `$` ended up being unnecessary, though if there are no other commits necessary then I'm not going to fix this and trigger re-execution of tests.

I was trying to change the port from 8080 to something else. I never got it to succeed. There is an 8080 somewhere else that I can't find.

**Pull request closed without merging**